ballista/rust/scheduler/src/state/executor_manager.rs

```diff
@@ -194,6 +200,33 @@ impl ExecutorManager {
         .await
     }
 
+    pub async fn get_client(
+        &self,
+        executor_id: &str,
+    ) -> Result<ExecutorGrpcClient<Channel>> {
+        let client = {
+            let clients = self.clients.read();
+            clients.get(executor_id).cloned()
+        };
+
+        if let Some(client) = client {
+            Ok(client)
+        } else {
+            let executor_metadata = self.get_executor_metadata(executor_id).await?;
+            let executor_url = format!(
+                "http://{}:{}",
+                executor_metadata.host, executor_metadata.grpc_port
+            );
+            let client = ExecutorGrpcClient::connect(executor_url).await?;
+
+            {
+                let mut clients = self.clients.write();
+                clients.insert(executor_id.to_owned(), client.clone());
+            }
+            Ok(client)
+        }
+    }
+
     /// Get a list of all executors along with the timestamp of their last recorded heartbeat
```

Comment:
There is a new method in utils.rs to create Grpc client which handles the grpc connection timeout and keep-alive settings, you can use that.

````
pub async fn create_grpc_client_connection<D>(
    dst: D,
) -> std::result::Result<Channel, Error>
where
    D: std::convert::TryInto<tonic::transport::Endpoint>,
    D::Error: Into<StdError>,
{
    let endpoint = tonic::transport::Endpoint::new(dst)?
        .connect_timeout(Duration::from_secs(20))
        .timeout(Duration::from_secs(20))
        // Disable Nagle's Algorithm since we don't want packets to wait
        .tcp_nodelay(true)
        .tcp_keepalive(Option::Some(Duration::from_secs(3600)))
        .http2_keep_alive_interval(Duration::from_secs(300))
        .keep_alive_timeout(Duration::from_secs(20))
        .keep_alive_while_idle(true);
    endpoint.connect().await
}